### PR TITLE
Branch one

### PR DESCRIPTION
**Pull request by Surabhi2011**

### CODE REVIEW

File: src/test/java/org/shl/testng/SHL_Prj1/TestScript.java [hunk 1]
```diff
@@ -20,6 +20,8 @@ public void Launch_Browser() throws Exception
   @Test
   public void Login()
   {
+
```
Comment (bhaskar211) [inline src/test/java/org/shl/testng/SHL_Prj1/TestScript.java]: no need

File: .settings/org.eclipse.jdt.core.prefs [hunk 2]
```diff
@@ -1,13 +1,13 @@
 eclipse.preferences.version=1
```
Comment (kartikbawa-prog) [inline .settings/org.eclipse.jdt.core.prefs:1]: remove this

File: src/main/java/Pages/Camera.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import utilities.BaseClass;
+import utilities.WaitExpectedConditions;
+
+public class Camera extends BaseClass{
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/Camera.java:11]: change Name of class as variables and functions are not specific to Camera.

File: src/main/java/Pages/Camera.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	WebDriver driver;
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/Camera.java:12]: make instance variables private

File: src/main/java/Pages/Camera.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	WebDriverWait wait;
+	
+	public Camera(WebDriver driver,WebDriverWait wait) {
+		super(driver,wait);
+		this.driver=driver;
+		this.wait=wait;
+		
+	}
+	public By Compare=By.xpath("//span[text()='COMPARE']");
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/Camera.java:21]: keep all instance variables together at top

File: src/main/java/Pages/Compare.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		this.wait=wait;
+	}
+	//Prints the product name and price with lowest price
+public void makeComparison(int size) {
+	int price=0;
+	int lowestprice=10000;
+	String LowProduct="";
+	String Product="";
+	for(int i=2;i<=size+1;i++) {
+		Product=driver.findElement(By.xpath("//*[@id=\"fk-compare-page\"]/div/div/div/div[1]/div[2]/div/div[1]/div[1]/div["+i+"]")).getText();
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/Compare.java:27]: Find better xpaths

File: src/main/java/Pages/Compare.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+		price=Integer.parseInt(driver.findElement(By.xpath("//div[2][@class='row']/div["+i+"]/div/div/div[1]")).getText().replace("₹","").replace(",", "").trim());
+	    if(price<lowestprice) {
+	    	lowestprice=price;
+	    	LowProduct=Product;
+	    }
+	}
+	System.out.println(LowProduct);
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/Compare.java:34]: use logs instead

File: src/main/java/Pages/HomePage.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import utilities.BaseClass;
+import Pages.Camera;
+import utilities.Constants;
+import utilities.WaitExpectedConditions;
+
+public class HomePage extends BaseClass {
+	public By CancelPopUp=By.xpath("//button[text()='✕']");
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/HomePage.java:20]: make them private

File: src/main/java/Pages/HomePage.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+	
+	public HomePage(WebDriver driver,WebDriverWait wait) {
+		super(driver,wait);
+		this.driver=driver;
+		this.wait=wait;
+		
+	}
+	//Cancels the modal box that pops up on site loading
+	public void CancelModal() {
+		try {
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/HomePage.java:39]: handle pop up properly

File: src/main/java/Pages/HomePage.java [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+import utilities.BaseClass;
+import Pages.Camera;
+import utilities.Constants;
+import utilities.WaitExpectedConditions;
+
+public class HomePage extends BaseClass {
+	public By CancelPopUp=By.xpath("//button[text()='✕']");
+	public By SendInfo=By.xpath("//input[@title='Search for products, brands and more']");
+	public By Search=By.xpath("//button[@type=\"submit\"]");
+	public By Electronics=By.xpath("//span[text()='Electronics']");
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/HomePage.java:23]: follow naming convention

File: src/main/java/Pages/ProductPage.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+		ArrayList<WebElement> list=new ArrayList<WebElement>();
+		list=(ArrayList<WebElement>) driver.findElements(By.xpath("//div[@data-id]"));
+		String ProductName;
+		int NumberOfRatings;
+		float starrating;
+		for(WebElement product:list) {
+			ProductName=product.findElement(By.xpath(".//a[@title]")).getText();
+
+			try{
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/ProductPage.java:60]: dont use try-catch as if-else

File: src/main/java/Pages/ProductPage.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+			}
+			try{
+				starrating=Float.parseFloat(product.findElement(By.className("hGSR34")).getText());
+			}
+			catch(Exception e) {
+				starrating=0;
+			}
+
+			if(NumberOfRatings==highestrating) {
+				System.out.println("Product Name: "+ProductName+"\nProduct Ratings: "+NumberOfRatings+"\nProduct StarRating: "+starrating);
```
Comment (kartikbawa-prog) [inline src/main/java/Pages/ProductPage.java:73]: logs

File: src/main/java/utilities/BaseClass.java [hunk 13]
```diff
@@ -22,15 +26,21 @@ public void click(WebElement link)
...
 	public void waitCondition(WaitExpectedConditions conditions,By locator,WebDriver driver,int timeOut) {
 	       wait=new WebDriverWait(driver,timeOut);
 	        switch (conditions){
 	        case PRESENCE_OF_ELEMENT:
 	                            wait.until(ExpectedConditions.presenceOfElementLocated(locator));
 	                            break;
 	        case ELEMENT_TO_BE_CLICKABLE: 
-	                          wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
-	                          break;    
+	                            wait.until(ExpectedConditions.visibilityOfElementLocated(locator));
```
Comment (kartikbawa-prog) [inline src/main/java/utilities/BaseClass.java:37]: change this

File: src/main/java/utilities/Initialiser.java [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+	public HomePage homePage;
+	public ProductPage productPage;
+	public Camera cameraPage;
+	public Compare comparePage;
+
+	
+   @BeforeClass
+    public void LaunchBrowser() {
+    	l_Browser=new Launcher();
+    	try {
```
Comment (kartikbawa-prog) [inline src/main/java/utilities/Initialiser.java:28]: handle exceptions properly

File: src/main/java/utilities/Product.java [hunk 15]
```diff
@@ -0,0 +1,25 @@
+package utilities;
```
Comment (kartikbawa-prog) [inline src/main/java/utilities/Product.java:1]: remove as it is not used

File: src/test/java/org/shl/testng/SHL_Prj1/TestScript2.java [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+import utilities.Initialiser;
+import utilities.WaitExpectedConditions;
+public class TestScript2 extends Initialiser
+{
+	@Test
+	public void TestPages() 
+	{    
+		homePage.CancelModal();
+		homePage.mouseHover();
+		cameraPage.addtoCompare(3);
```
Comment (kartikbawa-prog) [inline src/test/java/org/shl/testng/SHL_Prj1/TestScript2.java:19]: put hardcoded value in CONSTANTS

**Review by kartikbawa-prog — CHANGES_REQUESTED** [review]
Add better comments.
Take care of access modifiers.
and make other mentioned changes

File: src/main/java/Pages/HomePage.java [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+		
+	}
+	
+	//MouseHover over Electronics and then Compact & Bridge Cameras and clicks on it
+	public void mouseHover()
+	{
+		WebElement wb1=driver.findElement(Electronics);
+		WebElement wb2=driver.findElement(Camera);
+		
+		Actions act = new Actions(driver);
```
Comment (grajput380) [inline src/main/java/Pages/HomePage.java:59]: please create common method to use actions class

File: src/main/java/Pages/HomePage.java [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+		Action set1 = act.moveToElement(wb1).build();
+		set1.perform();
+		waitCondition(condition.VISIBILITY_OF,Camera,driver,10);
+		
+		Action set2 = act.moveToElement(wb2).click().build();
+		set2.perform();
+		waitCondition(condition.ELEMENT_TO_BE_CLICKABLE,CameraAcc,driver,10);
```
Comment (grajput380) [inline src/main/java/Pages/HomePage.java:66]: do we need to pass driver for this method ? I think driver is already available in BaseClass

File: src/main/java/utilities/Initialiser.java [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+    	driver=l_Browser.launchBrowser("Chrome");
+        driver.get(Constants.URL);}
+    	catch(Exception e) {
+    		System.out.println(e);
+    	}
+    	wait = new WebDriverWait(driver, 10);
+    	homePage=new HomePage(driver,wait);
+    	productPage=new ProductPage(driver,wait);
```
Comment (grajput380) [inline src/main/java/utilities/Initialiser.java:36]: please find a better approach to create object of pages

**Review by grajput380 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**